### ISSUE TITLE
Downloads keep going after switching to mobile data.

### ISSUE THREAD

**Issue by navid-zamani** [SkyTubeTeam/SkyTube#954]
On EMUI (Huawei) phones at least, a download started on wifi, keeps going, even when the wifi connection is lost. And even after disabling mobile data and re-enabling it.

Since, I think, this is the OS's download manager, that SkyTube only uses, which sadly, does not offer any configuration to the user, could you please detect switches to mobile data, and tell the download manager to stop downloading the videos? (And vice versa for when wifi comes back and Internet is available [As in: After the captive portal thing is done.]).

(Ideally, only pausing them, so they  can be resumed. Otherwise in certain situatioms, like on public transport with intermittend wifi at stations, downloads will never finish and keep going forever, draining the battery.)

That would be supernice for not draining all my mobile data. :)

**Comment by gzsombor** [comment]
Yes, you are right @navid-zamani SkyTube use Android's download manager, which has it's own bugs - and the code what calls it from SkyTube is very simplistic - for example doesn't really care, what happens if SkyTube stopped during download - if you close the app, the download wont show up (at least, if you request the download again, it detects that the file is there, so no need to re-download).
 Unfortunately I'm not aware of any free android download manager, what can be used easily. NewPipe use his own - which copied from somewhere else, and modified bit to integrate with NewPipe better - that can be forked, if someone has enough free time. But maybe there are other options too.
